family.pl

```diff
@@ -114,7 +125,15 @@
 %
 
 
-:- begin_tests(ancestor, [blocked('step 3')]).
+
+ancestor(X,Y) :- parent(X,Y);
+                 parent(X, E),
+                 ancestor(E,Y).
```

Comment:
While correct, it would probably be considered better style to split this into two clauses.